extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/configproperties/ConfigPropertiesBuildStep.java

```diff
@@ -89,7 +87,7 @@ void setup(CombinedIndexBuildItem combinedIndex,
                 boolean needsValidation = ClassConfigPropertiesUtil.addProducerMethodForClassConfigProperties(
                         deploymentClassLoader.getClassLoader(), classInfo, producerClassCreator,
                         configPropertiesMetadata.getPrefix(), configPropertiesMetadata.getNamingStrategy(),
-                        applicationIndex.getIndex(), configProperties);
```

Comment:
@geoand do you remember the reason why class-based `@ConfigProperties` requied the `ApplicationIndexBuildItem`? I think it's safe to use the `CombinedIndexBuildItem` here...

Reply:
Yeah, it's probably just an oversight on my part